rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -7257,24 +7260,9 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 			return;
 		}
 
-		const auto a = get_vr<f32[4]>(op.ra);
-		const auto b = get_vr<f32[4]>(op.rb);
-		const auto abs_a = fabs(a);
-		const auto abs_b = fabs(b);
-
-		// Actually, it's accurate and can be used as an alternative path for accurate xfloat.
-		if (g_cfg.core.spu_approx_xfloat)
-		{
-			// Compare abs values as integers, but return false if both are denormals or zeros.
-			const auto ia = bitcast<s32[4]>(abs_a);
-			const auto ib = bitcast<s32[4]>(abs_b);
-			const auto nz = eval((ia > 0x7fffff) | (ib > 0x7fffff));
-			set_vr(op.rt, sext<s32[4]>((ia > ib) & nz));
-		}
-		else
-		{
-			set_vr(op.rt, sext<s32[4]>(fcmp_ord(abs_a > abs_b)));
-		}
+                const auto a = eval(fabs(clamp_smax(get_vr<f32[4]>(op.ra))));
+                const auto b = eval(fabs(clamp_smax(get_vr<f32[4]>(op.rb))));
+                set_vr(op.rt, sext<s32[4]>(fcmp_ord(a > b)));
```

Comment:
accuracy of appprox path reduced.

Reply:
why?

Reply:
every +Nan is now equal to each other, same with -Nan.
this wasn't an issue before.

Reply:
From my observation we can only get NaN values if they are loaded immediately. e.g. 

```
il rt,0xffffffff -> -Smax
il rt,0x7fffffff -> +Smax
```
These would be clamped to smax and we are fine with that.

Reply:
bitcast<f32>(0x7fffffff) > bitcast<f32>(0x7ffffffe) for example breaks here.

Reply:
understood, but we can never create any of these numbers in the approximate xfloat path. On the other hand what to do with codings like these:

```
max = 0x7fffffff (CELL +Smax)
x = 1;
while (x < max) x = x * 2; ...
```

That would not terminate ... Both implementations will be valid in some special case ...